src/internal/operations/test/onedrive_test.go

```diff
@@ -911,6 +921,179 @@ func runDriveIncrementalTest(
 	}
 }
 
+var (
+	_ io.ReadCloser                    = &singleFileFailExtension{}
+	_ extensions.CreateItemExtensioner = &createSingleFileFailExtension{}
+)
+
+// singleFileFailExtension fails the first read on a file being uploaded during
+// a snapshot. Only one file is failed during the snapshot even if it the
+// snapshot contains multiple files.
+type singleFileFailExtension struct {
+	firstFile *atomic.Bool
```

Comment:
seems like a case for [sync.Once](https://pkg.go.dev/sync#Once)

Reply:
not sure if that'll make things any easier since the return from the `Read` function will be complicated. `Do` doesn't return anything so I have no way to tell if it ran or not and thus don't know if I should return an error or not in `Read`